src/Microsoft.Azure.WebJobs.Host/Bindings/QueueBindingRules.cs

```diff
@@ -0,0 +1,217 @@
+            converterManager.AddConverter<IStorageQueueMessage, string>(ConvertCloudQueueMessage2String);
+            converterManager.AddConverter<IStorageQueueMessage, byte[]>(ConvertCloudQueueMessage2ByteArray);
+
+            converterManager.AddConverter<CloudQueueMessage, IStorageQueueMessage>(ConvertReal2CloudQueueMessage);
+
+            var bf = new BindingFactory(nameResolver, converterManager);
+
+            var ruleQueueOutput = bf.BindToAsyncCollector<QueueAttribute, IStorageQueueMessage>(BuildFromQueueAttribute, ToParameterDescriptorForCollector, FixerUpper);
+            var ruleQueueClient = bf.BindToExactAsyncType<QueueAttribute, IStorageQueue>(BuildClientFromQueueAttributeAsync, ToParameterDescriptorForCollector, FixerUpper);
+            var ruleQueueClient2 = bf.BindToExactAsyncType<QueueAttribute, CloudQueue>(BuildRealClientFromQueueAttributeAsync, ToParameterDescriptorForCollector, FixerUpper);
+            var queueRules = new GenericCompositeBindingProvider<QueueAttribute>(
+                (attr) => ValidateQueueAttribute(attr, nameResolver),
+                ruleQueueClient, ruleQueueClient2, ruleQueueOutput);
+
+            return queueRules;
+        }
+
+        // Hook to apply hacky rules all at once. 
+        private async Task<QueueAttribute> FixerUpper(QueueAttribute attrResolved, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            // Look for [Storage] attribute and squerrel aover 
+            IStorageAccount account = await _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver);
+            StorageClientFactoryContext clientFactoryContext = new StorageClientFactoryContext
+            {
+                Parameter = parameter
+            };
+            IStorageQueueClient client = account.CreateQueueClient(clientFactoryContext);
+
+            // Create the queue if needed?
+            var queue = client.GetQueueReference(attrResolved.QueueName);
+            bool create = false;
+            if (create)
+            {
+                await queue.CreateIfNotExistsAsync(CancellationToken.None);
+            }
+
+            // Normalize and validate? 
+            string queueName = attrResolved.QueueName.ToLowerInvariant(); // must be lowercase. coerce here to be nice.
+            QueueClient.ValidateQueueName(queueName);
+
+            // Lowercase
+            return new ResolvedQueueAttribute(queueName, client);
+        }
+
+        private ParameterDescriptor ToParameterDescriptorForCollector(QueueAttribute attr, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            Task<IStorageAccount> t = Task.Run(() =>
+                _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver));
+            string accountName = t.GetAwaiter().GetResult().Credentials.AccountName;
+
+            return new QueueParameterDescriptor
```

Comment:
Does this parameter descriptor produce the exact same output as before? The reason I ask is that we have a bunch of Dashboard tests that cover the output. We'll need to run those.